receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows.go

```diff
@@ -155,6 +158,16 @@ func initializeDiskOpsMetric(metric pdata.Metric, startTime, now pdata.Timestamp
 func initializeDiskTimeMetric(metric pdata.Metric, startTime, now pdata.TimestampUnixNano, logicalDiskCounterValues []*perfcounters.CounterValues) {
 	diskTimeDescriptor.CopyTo(metric)
 
+	ddps := metric.DoubleSum().DataPoints()
+	ddps.Resize(len(logicalDiskCounterValues))
+	for idx, logicalDiskCounter := range logicalDiskCounterValues {
+		initializeDoubleDataPoint(ddps.At(idx), startTime, now, logicalDiskCounter.InstanceName, "", float64(now-startTime)/1e9-float64(logicalDiskCounter.Values[idleTime])/1e7)
```

Comment:
From https://docs.microsoft.com/en-us/archive/blogs/askcore/windows-performance-monitor-disk-counters-explained: "This counter [% Idle Time] provides a very precise measurement of how much time the disk remained in an idle state, meaning all the requests from the operating system to the disk have been completed and there are zero pending requests. The system timestamps an event when the disk goes idle, then timestamps another event when the disk receives a new request. At the end of the capture interval the percentage of the time spent in an idle state is calculated. This counter ranges from 100 (always idle) to 0 (always busy)."

So to calculate disk wall clock time we subtract the system uptime from the total sum of idle time. This pretty much works as expected but has one major caveat: the system doesn't record idle time while it's asleep. That means when the host wakes from sleep the entire time that it was asleep for would be recorded as busy time instead of idle time 😞. I suspect this probably wouldn't matter for Virtual Machines but I wonder if there are any other edge cases that would be reported weirdly if I do the computation this way.

Another alternative would be to record the time we start scraping & the idle time counter value at that time, and then just count from 0 every time the scraper restarts.

WDYT?

Reply:
If we report a monotonic sum with idleTime, the calculation that you make here can be done easily on the backend. I am not sure what this calculation gives us.

Reply:
The main reason is for consistency across Operating Systems.

For whatever reason:

Linux reports **disk io time**: time the disk was active
Windows reports **disk idle time**: time the disk was idle

So if we want to report the same metric for both we need to do some sort of conversion.

(aside: Windows does have a "Disk Time" perf counter but this measures something different)

Reply:
^ WDYT @jrcamp @asuresh4?

Reply:
Can we just report disk active then? On windows we will just do the diff not percentage.

Reply:
Sorry I think I read it wrong. You are doing just the difference

Reply:
Yea this new library I'm using reads the raw values used to compute perf counter values from the registry ==> when we get the value of a counter like "% Idle Time" we actually get "# seconds of idle time since boot" (with three caveat mentioned above)

Reply:
+1 on reporting the same metric across OSes. Should we also mention how the metric is calculated in the description since it's not a native measurement on Windows?

Reply:
> Maybe I need to add some comments to make this clear.

Added some package doc comments to the perfcounters package

> Should we also mention how the metric is calculated in the description since it's not a native measurement on Windows?

👍 I've updated the description to specify how this is calculated on Windows and I think that's good enough for now. We can review this in the future if we find any issues with the metric.